starfish/image/_stack.py

```diff
@@ -45,6 +52,23 @@ def numpy_array(self, data):
         self._data_needs_writeback = True
         data.setflags(write=False)
 
+    def slice(self, indices):
```

Comment:
This definitely needs a docstring. Looks like you expect a dict of 3-tuples with `start, stop, step` as indices? 

This is a bit clunky. I think at the very least we should be able to default to step=None since almost no one will want that (except, i guess, uniform sub-sampling?)

Reply:
+1 -- all this stack stuff should be over-documented, as it's the fundamental implementation of the input file format. from reading the code it's not  super obvious to me what's going on here

Reply:
Added docs and some simple examples.